lib/src/back_end/code_writer.dart

```diff
@@ -86,9 +86,6 @@ class CodeWriter {
   /// solution if the line ends up overflowing.
   final List<Piece> _currentUnsolvedPieces = [];
 
-  /// The options for the current innermost piece being formatted.
-  _PieceOptions get _options => _pieceOptions.last;
```

Comment:
Why did we remove this getter? I'd rather have a named variable than using `_options.last` below. Maybe just rename it?

Reply:
Prior to this change, we initialized `_pieceOptions` with a sort of "default" option with zero indentation. That way, the stack is never empty, and there is always a "parent" even for the outermost piece. It's just that the parent might be sort of fake. That avoided a couple of places where you otherwise need to do "If _pieceOptions isn't empty then use it, otherwise here's the default."

But with this change, `_PieceOptions` has a non-nullable reference to the `Piece` it was created for. That meant I couldn't initialize the stack with a fake outermost value because there's no piece to use for it. So instead, I just check to see if it's empty in the couple of places where it might be.

After that change, it felt weird to me to have an `_options` getter because it could throw if you happened to access at a time when the stack is empty. I thought it was a little clearer to just inline the `.last` calls everywhere so it was more obvious what was going on.